sycl/CMakeLists.txt

```diff
@@ -34,12 +34,12 @@ set(CLANG_VERSION "${CLANG_VERSION_MAJOR}.${CLANG_VERSION_MINOR}.${CLANG_VERSION
 set(LLVM_INST_INC_DIRECTORY "lib${LLVM_LIBDIR_SUFFIX}/clang/${CLANG_VERSION}/include")
 set(dst_dir ${LLVM_LIBRARY_OUTPUT_INTDIR}/clang/${CLANG_VERSION}/include)
 
-find_package(OpenCL)
+find_package(OpenCL 2.2)
```

Comment:
Alternative:
 - Add cmake option "USE_SYSTEM_OPENCL" with default value OFF.

1. By default we expect user to set `OpenCL_INCLUDE_DIRS` and `OpenCL_LIBRARIES`. 
2. If "USE_SYSTEM_OPENCL" is set to "ON", look for OpenCL in system paths.
3. If not found, try to download them from github.

Reply:
Why 'do you write the word 'Alternatively'.
find_package() also checks that OpenCL explicitly specified by OpenCL_INCLUDE_DIR and OpenCL_LIBRARY defs is 2.2 or newer. I.e. 2.2 is a reasonable check for now anyway. Right now SYCL headers are not compile-able with older OpenCL.

Regarding adding a new USE_SYSTEM_OPENCL option...
To me it does not seem really needed. 
Well, it may be good to have it only in this a+b scenario : a) there is OpenCL 2.2 in the system, b) even though it is 2.2 it not new enough, which causes fails.
Such hypothetical case can be easily worked around by manual download/build of OpenCL Headers and ICD-Loader.

I'll add it if you still insisting on having that USE_SYSTEM_OPENCL var.

Reply:
AFAIK, there are no OpenCL 2.2 implementations, but my point is that by default we use OpenCL installed in the system, which we probably shouldn't do as we can't guarantee that this configuration will work.
This is why I asked to move `find_package` under disabled cmake option.
By default we should build SYCL runtime with Khronos ICD loader.

Reply:
Ok, I agree, and did update of this patch.
Now this patch includes your patch https://github.com/intel/llvm/pull/591